### PR TITLE
whisper: remove whisper

### PR DESCRIPTION
**Pull request by gballet**
Remove Whisper. Rework #21371 after the node refactor.

### CODE REVIEW

File: cmd/geth/config.go [hunk 1]
```diff
@@ -137,13 +138,12 @@ func makeConfigNode(ctx *cli.Context) (*node.Node, gethConfig) {
 }
 
 // enableWhisper returns true in case one of the whisper flags is set.
-func enableWhisper(ctx *cli.Context) bool {
+func checkWhisper(ctx *cli.Context) {
 	for _, flag := range whisperFlags {
 		if ctx.GlobalIsSet(flag.GetName()) {
-			return true
+			log.Warn("deprecated whisper flag detected. Whisper has been moded to github.com/ethereum/whisper")
```
Comment (MariusVanDerWijden) [inline cmd/geth/config.go:144]:
```suggestion
			log.Warn("deprecated whisper flag detected. Whisper has been moved to github.com/ethereum/whisper")
```

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM (nitpick)

Question: Did we also move the fuzzer to ethereum/whisper?

File: cmd/geth/usage.go [hunk 2]
```diff
@@ -211,7 +211,7 @@ var AppHelpFlagGroups = []flags.FlagGroup{
 		Flags: metricsFlags,
 	},
 	{
-		Name:  "WHISPER (EXPERIMENTAL)",
+		Name:  "WHISPER (deprecated)",
```
Comment (renaynay) [inline cmd/geth/usage.go:214]: Maybe move this under the `show-deprecated-flags` command since the functionality is deprecated.

**Review by renaynay — APPROVED** [review]

**Comment by gballet** [comment]
> Question: Did we also move the fuzzer to ethereum/whisper?

Yes

**Comment by holiman** [comment]
Two suggestions, to remove the whisper dependency
```diff
diff --git a/cmd/geth/config.go b/cmd/geth/config.go
index ce490319a1..2c15a4c832 100644
--- a/cmd/geth/config.go
+++ b/cmd/geth/config.go
@@ -32,7 +32,6 @@ import (
        "github.com/ethereum/go-ethereum/log"
        "github.com/ethereum/go-ethereum/node"
        "github.com/ethereum/go-ethereum/params"
-       "github.com/ethereum/whisper"
        "github.com/naoina/toml"
 )
 
@@ -76,7 +75,6 @@ type ethstatsConfig struct {
 
 type gethConfig struct {
        Eth      eth.Config
-       Shh      whisper.Config
        Node     node.Config
        Ethstats ethstatsConfig
 }
@@ -111,7 +109,6 @@ func makeConfigNode(ctx *cli.Context) (*node.Node, gethConfig) {
        // Load defaults.
        cfg := gethConfig{
                Eth:  eth.DefaultConfig,
-               Shh:  whisper.DefaultConfig,
                Node: defaultNodeConfig(),
        }
 
@@ -132,7 +129,7 @@ func makeConfigNode(ctx *cli.Context) (*node.Node, gethConfig) {
        if ctx.GlobalIsSet(utils.EthStatsURLFlag.Name) {
                cfg.Ethstats.URL = ctx.GlobalString(utils.EthStatsURLFlag.Name)
        }
-       utils.SetShhConfig(ctx, stack, &cfg.Shh)
+       utils.SetShhConfig(ctx, stack)
 
        return stack, cfg
 }
```
and 
```diff
diff --git a/cmd/utils/flags.go b/cmd/utils/flags.go
index 2fcb46ecb1..731df9c17b 100644
--- a/cmd/utils/flags.go
+++ b/cmd/utils/flags.go
@@ -63,7 +63,6 @@ import (
 	"github.com/ethereum/go-ethereum/p2p/nat"
 	"github.com/ethereum/go-ethereum/p2p/netutil"
 	"github.com/ethereum/go-ethereum/params"
-	"github.com/ethereum/whisper"
 	pcsclite "github.com/gballet/go-libpcsclite"
 	cli "gopkg.in/urfave/cli.v1"
 )
@@ -639,12 +638,12 @@ var (
 	WhisperMaxMessageSizeFlag = cli.IntFlag{
 		Name:  "shh.maxmessagesize",
 		Usage: "Max message size accepted",
-		Value: int(whisper.DefaultMaxMessageSize),
+		Value: 1024 * 1024,
 	}
 	WhisperMinPOWFlag = cli.Float64Flag{
 		Name:  "shh.pow",
 		Usage: "Minimum POW accepted",
-		Value: whisper.DefaultMinimumPoW,
+		Value: 0.2,
 	}
 	WhisperRestrictConnectionBetweenLightClientsFlag = cli.BoolFlag{
 		Name:  "shh.restrict-light",
@@ -1465,7 +1464,7 @@ func CheckExclusive(ctx *cli.Context, args ...interface{}) {
 }
 
 // SetShhConfig applies shh-related command line flags to the config.
-func SetShhConfig(ctx *cli.Context, stack *node.Node, cfg *whisper.Config) {
+func SetShhConfig(ctx *cli.Context, stack *node.Node) {
 	if ctx.GlobalIsSet(WhisperMaxMessageSizeFlag.Name) || ctx.GlobalIsSet(WhisperMinPOWFlag.Name) || ctx.GlobalIsSet(WhisperRestrictConnectionBetweenLightClientsFlag.Name) {
 		log.Warn("Whisper support has been deprecated and the code has been moved to github.com/ethereum/whisper")
 	}
```

**Comment by holiman** [comment]
I don't suppose we can also remove the whisper dep from `mobile/types.go`? Is that supposed to be able to talk to a standalone whisper node which isn't part of geth?

File: cmd/geth/usage.go (hunk 2 above)
Comment (holiman) [reply cmd/geth/usage.go:214]: ^ this?

**Review by holiman — APPROVED** [review]
LGTM

**Comment by Lucienest** [comment]
What's the motivation behind removing the Whisper?
@holiman

**Comment by holiman** [comment]
Well, when we removed it, @gballet was the primary maintainer, so he's better suited to answer this question than I am.

**Comment by karalabe** [comment]
@Lucienest Whisper as implemented in go-ethereum is a non-feasible protocol, there was nobody to maintain it and Status are working on an alternative with a full team. Without an active research team, Whisper cannot work, and we ourselves can't focus on Ethereum and Whisper simultaneously.

**Pull request merged**